automation_tools/satellite6/upgrade/client.py

```diff
@@ -0,0 +1,115 @@
+        execute(disable_repos, old_repo, host=client)
+        print('Upgrading Clients:')
+        execute(lambda: run('yum update katello-agent'), host=client)
+        print('katello-agent package after installation:')
+        execute(lambda: run('rpm -qa | grep katello-agent'), host=client)
+
+
+def docker_clients_upgrade(old_repo, container_ids):
+    """Helper function to run upgrade on docker containers as clients
+
+    :param string old_repo: The old tools repo to disable before updating
+        katello-agent package
+    :param list container_ids: The list of container_ids onto which
+        katello-agent package will be updated
+    """
+    for client in container_ids:
+        print('Upgrading client on docker container: {}'.format(client))
+        print('Previous katello-agent version:')
+        docker_execute_command(client, 'rpm -qa | grep katello-agent')
+        print('Disabling existing client tools repo:')
+        docker_execute_command(
+            client, 'subscription-manager repos --disable {}'.format(old_repo))
+        print('Upgrading clients on Docker Containers')
+        docker_execute_command(client, 'yum update katello-agent')
+        print('katello-agent package after installation:')
+        docker_execute_command(client, 'rpm -qa | grep katello-agent')
+
+
+def satellite6_client_setup():
+    """Sets up required things on upgrade running machine and on Client to
+    perform client upgrade later
+
+    If not personal clients, then it creates docker containers as clients on
+    rhevm vm.
+    """
+    # If Personal Client Hostname provided
+    if os.environ.get('CLIENT6_HOSTS') or os.environ.get('CLIENT7_HOSTS'):
+        clients6 = list(os.environ.get('CLIENT6_HOSTS'))
+        clients7 = list(os.environ.get('CLIENT7_HOSTS'))
+        # Sync latest sat tools repo to clients if downstream
+        if all([
+            os.environ.get('TOOLS_URL_RHEL6'),
+            os.environ.get('TOOLS_URL_RHEL7')
+        ]):
+            sync_tools_repos_to_upgrade('rhel6', clients6)
+            sync_tools_repos_to_upgrade('rhel7', clients7)
+    # Run upgrade on Docker Containers
+    else:
+        docker_vm = os.environ.get('DOCKER_VM')
+        clients_count = os.environ.get('CLIENTS_COUNT')
+        # First delete if any containers running
+        remove_all_docker_containers(only_running=False)
+        # Generate Clients on RHEL 7 and RHEL 6
+        clients6 = execute(generate_satellite_docker_clients_on_rhevm(
+            'rhel6', int(clients_count)/2), host=docker_vm)[docker_vm]
+        clients7 = execute(generate_satellite_docker_clients_on_rhevm(
+            'rhel7', int(clients_count)/2), host=docker_vm)[docker_vm]
+        # Sync latest sat tools repo to clients if downstream
+        if all([
+            os.environ.get('TOOLS_URL_RHEL6'),
+            os.environ.get('TOOLS_URL_RHEL7')
+        ]):
+            sync_tools_repos_to_upgrade('rhel6', clients7.keys())
```

Comment:
Sorry I made the change in User defined(personal) clients as if rhel6_clients and if rhel7_clienrts are given.

But the code you pointed here is installing clients from Docker image as containers. And the user has no choice to choose either rhel6 or rhel7. The user can only specify the nuber of clients he want and that number will be devided by 2. E.g if user said he want 50 clients the 25 clients of rhel6 and 25 clients of rhel7 will be created :)